src/cosmosdb-preview/azext_cosmosdb_preview/custom.py

```diff
@@ -604,6 +218,63 @@ def cli_cosmosdb_managed_cassandra_datacenter_update(client, resource_group_name
     return client.begin_create_update(resource_group_name, cluster_name, data_center_name, data_center_resource)
 
 
+def _handle_exists_exception(http_response_error):
+    if http_response_error.status_code == 404:
+        return False
+    raise http_response_error
+
+
+def cli_cosmosdb_graph_create(client, account_name, resource_group_name, graph_name):
+    graph = GraphResourceCreateUpdateParameters(resource=GraphResource(id=graph_name))
+    return client.begin_create_update_graph(resource_group_name, account_name, graph_name, graph)
+
+
+def cli_cosmosdb_graph_exists(client, account_name, resource_group_name, graph_name):
+    try:
+        client.get_graph(resource_group_name, account_name, graph_name)
+    except HttpResponseError as ex:
+        return _handle_exists_exception(ex)
+
+    return True
+
+
+def cli_cosmosdb_service_exists(client, account_name, resource_group_name, service_name):
+    try:
+        client.get(resource_group_name, account_name, service_name)
+    except HttpResponseError as ex:
+        return _handle_exists_exception(ex)
+
+    return True
+
+
+def cli_cosmosdb_service_create(client,
+                                account_name,
+                                resource_group_name,
+                                service_kind,
+                                service_name,
+                                instance_count,
+                                instance_size):
```

Comment:
Are all these parameters required? Do they have default values?

Reply:
I can add default values to instance_Count and instance_size, others cannot have defaults

Reply:
Added for create scenario, not required for update scenario